cvat/apps/analytics_report/report/primary_metrics/imetric.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2023 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from cvat.apps.analytics_report.report.primary_metrics.utils import make_clickhouse_query
+from datetime import datetime, timezone
+
+class IPrimaryMetric():
```

Comment:
I suppose all these classes prefixed with `I` to reflect they describe an interface, but they describe abstract classes instead by providing an implementation and defining the fields. If you only wanted to declare fields, it will be better to change declarations to be like this:

```python
class MyClass:
    field: type
```

This will define just an interface with no implementation. For abstract base classes it would be meaningful to add the [`ABCMeta` metaclass](https://docs.python.org/3.10/library/abc.html) (so the class methods that are supposed for redefining were required to be redefined in subclasses):

```
from abc import ABCMeta, abstractmethod

class MyClass(metaclass=ABCMeta):
    @abstractmethod
    def foo(self):
        ... # must be redefined in a subclass
```